service/controller.go

```diff
@@ -199,155 +227,285 @@ func (s *service) CreateVolume(
 		}
 	}
 
-	params = mergeStringMaps(params, req.GetSecrets())
-
-	// We require the storagePool name for creation
-	sp, ok := params[KeyStoragePool]
-	if !ok {
-		return nil, status.Errorf(codes.InvalidArgument,
-			"%s is a required parameter", KeyStoragePool)
+	if req.VolumeCapabilities[0].GetBlock() != nil {
+		// We need to check if user requests raw block access from nfs and prevent that
+		fsType, ok := params[KeyFsType]
+		// FsType can be empty
+		if ok && fsType == "nfs" {
+			return nil, status.Errorf(codes.InvalidArgument, "raw block requested from NFS Volume")
+		}
 	}
 
-	pdID := ""
-	pd, ok := params[KeyProtectionDomain]
-	if !ok {
-		Log.Printf("Protection Domain name not provided; there could be conflicts if two storage pools share a name")
-	} else {
-		pdID, err = s.getProtectionDomainIDFromName(systemID, pd)
+	nfsAcls := s.opts.NfsAcls
+	var arr *ArrayConnectionData
+	sysID := s.opts.defaultSystemID
+	arr = s.opts.arrays[sysID]
+
+	if isNFS {
+		// fetch NAS server ID
+		nasName, ok := params[KeyNasName]
+		if !ok {
+			return nil, status.Errorf(codes.InvalidArgument, "`%s` is a required parameter", KeyNasName)
+		}
+		nasServerID, err := s.getNASServerIDFromName(systemID, nasName)
 		if err != nil {
 			return nil, err
 		}
-	}
 
-	volType := s.getVolProvisionType(params) // Thick or Thin
-
-	name := req.GetName()
-	if name == "" {
-		return nil, status.Error(codes.InvalidArgument,
-			"Name cannot be empty")
-	}
+		// fetch storage pool ID
+		pdID := ""
+		pd, ok := params[KeyProtectionDomain]
+		if !ok {
+			Log.Printf("Protection Domain name not provided; there could be conflicts if two storage pools share a name")
+		} else {
+			pdID, err = s.getProtectionDomainIDFromName(systemID, pd)
+			if err != nil {
+				return nil, err
+			}
+		}
 
-	if len(name) > 31 {
```

Comment:
Can we have this name check for nfs volumes also.

Reply:
done